samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/MainDialog.java

```diff
@@ -0,0 +1,238 @@
+import com.microsoft.bot.dialogs.DialogTurnResult;
+import com.microsoft.bot.dialogs.WaterfallDialog;
+import com.microsoft.bot.dialogs.WaterfallStep;
+import com.microsoft.bot.dialogs.WaterfallStepContext;
+import com.microsoft.bot.dialogs.prompts.PromptOptions;
+import com.microsoft.bot.dialogs.prompts.TextPrompt;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.microsoft.bot.builder.BotTelemetryClient;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * The class containing the main dialog for the sample.
+ */
+public class MainDialog extends ComponentDialog {
+
+    private final FlightBookingRecognizer luisRecognizer;
+    private final Integer plusDayValue = 7;
+
+    /**
+     * The constructor of the Main Dialog class.
+     *
+     * @param withLuisRecognizer  The FlightBookingRecognizer object.
+     * @param bookingDialog       The BookingDialog object with booking dialogs.
+     * @param withTelemetryClient The BotTelemetryClient to use.
+     */
+    public MainDialog(FlightBookingRecognizer withLuisRecognizer, BookingDialog bookingDialog, BotTelemetryClient withTelemetryClient) {
+        super("MainDialog");
+
+        luisRecognizer = withLuisRecognizer;
+        
+        // Set the telemetry client for this and all child dialogs
+        this.setTelemetryClient(withTelemetryClient);
+
+        addDialog(new TextPrompt("TextPrompt"));
+        addDialog(bookingDialog);
+        WaterfallStep[] waterfallSteps = {
+            this::introStep,
+            this::actStep,
+            this::finalStep
+        };
+        addDialog(new WaterfallDialog("WaterfallDialog", Arrays.asList(waterfallSteps)));
+
+        // The initial child Dialog to run.
+        setInitialDialogId("WaterfallDialog");
+    }
+
+
+    /**
+     * First step in the waterfall dialog. Prompts the user for a command. Currently, this expects a
+     * booking request, like "book me a flight from Paris to Berlin on march 22" Note that the
+     * sample LUIS model will only recognize Paris, Berlin, New York and London as airport cities.
+     *
+     * @param stepContext A {@link WaterfallStepContext}
+     * @return A {@link DialogTurnResult}
+     */
+    private CompletableFuture<DialogTurnResult> introStep(WaterfallStepContext stepContext) {
+        if (!luisRecognizer.isConfigured()) {
+            Activity text = MessageFactory.text("NOTE: LUIS is not configured. "
+                + "To enable all capabilities, add 'LuisAppId', 'LuisAPIKey' and 'LuisAPIHostName' "
+                + "to the appsettings.json file.", null, InputHints.IGNORING_INPUT);
```

Comment:
```suggestion
                + "to the application.properties file.", null, InputHints.IGNORING_INPUT);
```

This is wrong also in the [corebot](https://github.com/microsoft/botbuilder-java/blob/main/samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java) 🤦🏾‍♂️ 

Let's fix it in both places (also in the template of the generator)